### PR TITLE
Add security-group by tag python rule

### PR DESCRIPTION
**Pull request by aaronkhoo**
Add a new python rule that ensures all EC2 Instances that match a certain tag regex format also have a specific security group attached.

### CODE REVIEW

**Comment by pprahlad** [comment]
Would recommend "TagName" as the rule parameter because name can be confused with the instance name

**Comment by aaronkhoo** [comment]
@pprahlad Actually I am targeting the instance name specifically (which shows up as a tag with key value "name").

File: python/ec2_require_security_group_by_tag.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+# 
+
+import boto3
+import json
+import re
+
+def is_applicable(config_item, event):
+    status = config_item['configurationItemStatus']
+    event_left_scope = event['eventLeftScope']
+    test = ((status in ['OK', 'ResourceDiscovered']) and
```
Comment (ryflood) [inline python/ec2_require_security_group_by_tag.py]: variable name 'test' is a bit vague. Please consider an alternative, or simply return from this line

File: python/ec2_require_security_group_by_tag.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            event_left_scope == False)
+    return test
+
+def evaluate_compliance(config_item, rule_parameters):
+    # Initialize evaluation to 'not applicable', i.e. rule doesn't apply
+    evaluation = 'NOT_APPLICABLE'
+    if (config_item['resourceType'] == 'AWS::EC2::Instance'):
```
Comment (ryflood) [inline python/ec2_require_security_group_by_tag.py]: this check may make more sense in the 'is_applicable' method

File: python/ec2_require_security_group_by_tag.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        configuration = config_item['configuration']
+        tags = configuration['tags']
+        foundTag = False
```
Comment (ryflood) [inline python/ec2_require_security_group_by_tag.py]: is this variable required?

File: python/ec2_require_security_group_by_tag.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        # If the config item is for an EC2 instance, then iterate through the tags for that instance
+        for tag in tags:
+            # Check if this is the name tag, and that it matches the provided regex value
```
Comment (ryflood) [inline python/ec2_require_security_group_by_tag.py]: [nit] - "Check if this is the 'Name' tag..." or "Check if this tag has key 'Name'..." may be more readable

**Pull request merged**